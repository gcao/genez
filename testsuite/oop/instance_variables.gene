#!/usr/bin/env gene run

# Test: Instance variable declarations with (var /name)

(class Person
  (var /first_name)
  (var /last_name)
  (var /age))

(class Book
  (var /title)
  (var /author)
  (var /isbn)
  [pages 0])

# Mixed field declarations
(class Product
  (var /id)
  [name "Unknown"]
  (var /price)
  [in_stock true])

(print "Instance variable declarations parsed successfully")