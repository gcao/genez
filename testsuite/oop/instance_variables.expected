Instance variable declarations parsed successfully